### PR TITLE
style(Popup): fix tests, update typings and propTypes usage

### PR DESCRIPTION
**Pull request by layershifter**
This PR is part of work for removing propTypes in production bundle (#524, #731).
Also, cleanups and updates typings for #1072.
Also, fixes `positions` test.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -11,7 +11,6 @@ export type SemanticSIZES = 'mini' | 'tiny' | 'small' | 'medium' | 'large' | 'bi
...
-export type SemanticPOSITIONING = 'top left' | 'top right' | 'bottom right' | 'bottom left' | 'right center' | 'left center' | 'top center' | 'bottom center';
```
Comment (layershifter) [inline src/index.d.ts:14]: This type used only by `Popup`.

File: src/modules/Popup/Popup.js [hunk 2]
```diff
@@ -17,25 +19,16 @@ import PopupHeader from './PopupHeader'
...
+export const POSITIONS = [
+  'top left',
+  'top right',
+  'bottom right',
+  'bottom left',
+  'right center',
+  'left center',
+  'top center',
+  'bottom center',
+]
```
Comment (layershifter) [inline src/modules/Popup/Popup.js:31]: Positions are used in multiple places, so we need it as const.

File: src/modules/Popup/Popup.js [hunk 3]
```diff
@@ -349,8 +349,8 @@ export default class Popup extends Component {
     const popupJSX = (
       <ElementType {...rest} className={classes} style={style} ref={this.popupMounted}>
         {children}
-        {!children && PopupHeader.create(header)}
-        {!children && PopupContent.create(content)}
+        {_.isNil(children) && PopupHeader.create(header)}
+        {_.isNil(children) && PopupContent.create(content)}
```
Comment (layershifter) [inline src/modules/Popup/Popup.js:353]: Our problem with '0' fixed there 😄

File: test/specs/modules/Popup/Popup-test.js [hunk 4]
```diff
@@ -107,23 +107,20 @@ describe('Popup', () => {
   })
 
   describe('positioning', () => {
-    it('is always within the viewport', () => {
-      _.each(Popup._meta.props.positions, position => {
+    POSITIONS.forEach(position => {
+      it('is always within the viewport', () => {
```
Comment (layershifter) [inline test/specs/modules/Popup/Popup-test.js:111]: This test work now and #1239 will work.

There is simple explanation why it worked before: test used `_meta.props.positions` while they were defined in `_meta.props.positioning`, so it doesn't really work. But, when I add real positions, it failed. `beforeEach` handler applied to `wrapperMount` once, now it applies on each `position` and everything works fine.

File: test/specs/modules/Popup/Popup-test.js [hunk 5]
```diff
@@ -265,13 +262,10 @@ describe('Popup', () => {
   })
 
   describe('size', () => {
-    it('defines prop options in _meta', () => {
-      Popup._meta.props.should.have.any.keys('size')
-      Popup._meta.props.size.should.be.an('array')
-    })
```
Comment (layershifter) [inline test/specs/modules/Popup/Popup-test.js:271]: Cruft.

File: test/specs/modules/Popup/PopupContent-test.js [hunk 6]
```diff
@@ -4,4 +4,6 @@ import * as common from 'test/specs/commonTests'
 describe('PopupContent', () => {
   common.isConformant(PopupContent)
   common.rendersChildren(PopupContent)
+
+  common.implementsCreateMethod(PopupContent)
```
Comment (layershifter) [inline test/specs/modules/Popup/PopupContent-test.js:8]: We need `commonTest` there :)

File: test/specs/modules/Popup/PopupHeader-test.js [hunk 7]
```diff
@@ -4,4 +4,6 @@ import * as common from 'test/specs/commonTests'
 describe('PopupHeader', () => {
   common.isConformant(PopupHeader)
   common.rendersChildren(PopupHeader)
+
+  common.implementsCreateMethod(PopupHeader)
```
Comment (layershifter) [inline test/specs/modules/Popup/PopupHeader-test.js:8]: And there :)

**Review by levithomason — APPROVED** [review]
:boom:

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.65.0`

**Pull request merged**